### PR TITLE
Updated editor.rs, to address issue #459

### PR DESCRIPTION
**Pull request by Cristian-Andrei-Pomenea**
Modified the search function to start from cursor position instead of the start, this required an additional change in the calculations of "current", to avoid jumping down each time a new char is added.

### CODE REVIEW

File: src/editor.rs [hunk 1]
```diff
@@ -647,10 +647,19 @@ impl Editor {
     fn find(&mut self, query: &str, last_match: Option<usize>, forward: bool) -> Option<usize> {
         // Number of rows to search
         let num_rows = if query.is_empty() { 0 } else { self.rows.len() };
-        let mut current = last_match.unwrap_or_else(|| num_rows.saturating_sub(1));
+        let mut current = last_match.unwrap_or(self.cursor.y);
+        // Some edge cases fail without this, as seen from the tests
+        current = current.min(self.rows.len().saturating_sub(1));
```
Comment (ilai-deutel) [inline src/editor.rs]: This can fit in one line:
```suggestion
        let mut current = last_match.unwrap_or(self.cursor.y).min(num_rows.saturating_sub(1));
```

**Review by ilai-deutel — CHANGES_REQUESTED** [review]
Thank you for the PR! This is working well!
